### DISCUSSION TITLE
Foxglove agent option to ignore bags actively recording

### DISCUSSION THREAD

**Discussion by tonynajjar** [foxglove/community#986] [General]
Foxglove agent currently only allows to ignore file based on their name, however rosbag2 currently does not offer a way to mark active bag files with e.g. a suffix. When foxglove agent tries to ingest a bag being recorded it throws the error:

```
{"message":"Failed to ingest rosbag2_2024_04_11-10_17_14/rosbag2_2024_04_11-10_17_14_0.mcap: failed to summarize input rosbag2_2024_04_11-10_17_14/rosbag2_2024_04_11-10_17_14_0.mcap: failed to read magic: EOF"
``` 

It would be nice if Foxglove agent had an extra option to ignore bags being recorded, in other words "open files" (e.g. with lsof)

Related discussion: https://github.com/ros2/rosbag2/issues/1597

**Comment by defunctzombie** [comment]
FYI with agent v1.1.4 we've added new logic that will ignore incomplete mcap files in the watched directory: https://docs.foxglove.dev/changelog/agent/v1.1.4

This should reduce the error spam you are seeing.

**Comment by tonynajjar** [comment]
Yeah thanks for that. "Reduce" is the keyword though. It still produces continuous logs that it's skipping the files though. It would have been nice to even skip these logs.

**Reply by defunctzombie** [reply]
@james-rms fyi
